mirage/client/dns_client_mirage.ml

```diff
@@ -1,12 +1,69 @@
 open Lwt.Infix
 
+let ( let+ ) = Result.bind
+
+let inet_addr_of_string str =
+  let lst = String.split_on_char ':' str in
+  let int_of_string_opt str = try Some (int_of_string str) with _ -> None in
+  ( match List.rev lst with
+  | [ ipv4 ] ->
+    let+ ipaddr = Ipaddr.of_string ipv4 in Ok (ipaddr, 53)
+  | port :: ipaddr ->
+    ( match Ipaddr.of_string (String.concat ":" (List.rev ipaddr)),
+            Ipaddr.of_string (String.concat ":" lst),
+            int_of_string_opt port with
+    | _, Ok ipaddr, _ -> Ok (ipaddr, 53)
+    | Ok ipaddr, Error _, Some port -> Ok (ipaddr, port)
+    | _, Error _, _ -> Error (`Msg (Fmt.str "Invalid inet-addr: %S" str)) )
+  | [ ] -> Error (`Msg "Invalid empty inet-addr") ) 
```

Comment:
Sorry for being picky, but to reuse this functionality, I'd prefer to have it in the ocaml-ipaddr package. Maybe a `with_port_of_string : ?default:int -> string -> (t * port, [> `Msg of string ]) result` in all three modules? :)

Reply:
> in all three modules?

Which module you want? You would like to integrate that into `ipaddr` (like `Ipaddr.with_port_of_string`)?

Reply:
Yes, that would be great imho.